packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +222,112 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
   }
 }
 
+/// Popup menu item for the product list page: export list.
+class ProductListPopupExport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName =
+        "${listName.replaceAll(' ', '-').toLowerCase()}_${DateTime.now().millisecondsSinceEpoch}.csv";
+    final StringBuffer csv = StringBuffer('Barcode,Name,Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '').replaceAll(';', '');
+        final String brand = (product.brands ?? '').replaceAll(';', '');
+        csv.write('"$barcode";"$name";"$brand"\n');
+      }
+    }
+
+    unawaited(
+      Share.shareXFiles(
+        <XFile>[
+          XFile.fromData(
+            utf8.encode(csv.toString()),
+            name: '$fileName.csv',
+            mimeType: 'text/csv',
+          ),
+        ],
+        fileNameOverrides: <String>['$fileName.csv'],
+      ),
+    );
+
+    return null;
+  }
+}
+
+/// Popup menu item for the product list page: import list.
+class ProductListPopupImport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_import;
+
+  @override
+  IconData getIconData() => Icons.upload;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.import;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: <String>['csv'],
+    ).then((final FilePickerResult? result) {
```

Comment:
Not a big fan of the `then` syntax, and in that case particularly as we may do something different if we really imported data.
Please consider a classical `await` syntax instead.